### ISSUE TITLE
v1.6.0 BlurView No Longer Works on Emulator

### ISSUE THREAD

**Issue by searchy2** [Dimezis/BlurView#93]
Thank you for updating the library and continuously improving the features!

I am trying to upgrade my libraries to use v1.6.0 but I run into an issue where the BlurView doesn't render because the new BlurViewFacade might be interfering. 

v1.6.0
![MaxthonSnap20190417215002](https://user-images.githubusercontent.com/19896174/56333843-61120580-615b-11e9-8a63-07e15a6c4071.png)

v1.5.0 (working)
![MaxthonSnap20190417215410](https://user-images.githubusercontent.com/19896174/56333847-65d6b980-615b-11e9-9f6e-23d2a626b5bf.png)

I'm extending BlurView because I need to apply rounded corners to the blur. This is my custom BlurView with rounded corners:

```
package stream.customalert.ui;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewOutlineProvider;

import eightbitlab.com.blurview.BlurView;
import eightbitlab.com.blurview.RenderScriptBlur;
import stream.customalert.CustomAlertDialogue;

public class CustomBlurDialogue extends BlurView {

    public CustomBlurDialogue(Context context) {
        super(context);
        init();
    }

    public CustomBlurDialogue(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
        initCorners(context);
    }

    public CustomBlurDialogue(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
        initCorners(context);
    }

    private void init() {

    }

    private void initCorners(Context context) {
            setRoundedCorners(CustomAlertDialogue.Units.dpToPx(context, 15));
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
    }

    public void create(View decorView, float radius) {

        ViewGroup rootView = decorView.findViewById(android.R.id.content);
        Drawable windowBackground = decorView.getBackground();

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
            setupWith(rootView)
                    .setFrameClearDrawable(windowBackground)
                    .setBlurAlgorithm(new RenderScriptBlur(getContext()))
                    .setBlurRadius(radius)
                    .setHasFixedTransformationMatrix(true);
        }
    }

    /**
     * Set Rounded Corners on Lollipop and above. Use rounded drawable and disable blur below Lollipop.
     * @param cornerRadius - set corner radius in pixels.
     */
    private void setRoundedCorners(int cornerRadius) {

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            setBackground(new RoundedCornersDrawable(cornerRadius));
            setOutlineProvider(ViewOutlineProvider.BACKGROUND);
            setClipToOutline(true);
        }
    }
}
```

Before the introduction of BlurViewFacade, it was working because the blur was handled within the BlurView class. Now, it appears the blur is handled by the BlurViewFacade. Do you have any ideas how I can extend BlurView successfully? 

Perhaps there is a way to have BlurViewFacade without breaking BlurView's extensibility or maybe I'm using BlurViewFacade wrong?

Thanks for taking a look!

**Comment by Dimezis** [comment]
Weird, your setup works fine for me.
Don't know what's your `RoundedCornersDrawable` implementation, so I made my own:

```XML
<?xml version="1.0" encoding="utf-8"?>
<shape xmlns:android="http://schemas.android.com/apk/res/android"
    android:shape="rectangle">
    <corners android:radius="20dp" />
</shape>
```

Please post your `RoundedCornersDrawable` (I guess this might be the reason), phone, OS, and whether you're using hardware acceleration or not.

**Comment by searchy2** [comment]
Thanks for the quick reply!

Sorry for not including my `RoundedCornersDrawable`, I didn't mean to make it harder for you to help. Here's the code:
```
package stream.customalert.ui;

import android.graphics.Color;
import android.graphics.drawable.GradientDrawable;

public class RoundedCornersDrawable extends GradientDrawable {

    public RoundedCornersDrawable(int cornerRadius){
        super(Orientation.BOTTOM_TOP, new int[]{Color.TRANSPARENT, Color.TRANSPARENT, Color.TRANSPARENT});
        setCornerRadius(cornerRadius);
    }
}
```

All the library files can be found in my CustomAlertViewDialogue library https://github.com/searchy2/CustomAlertViewDialogue

When I tried loading the BlurView on a physical device vs on an emulator, It worked perfectly! So it seems like hardware acceleration is the issue. Why would previous versions of BlurView work on an emulator but not the latest? I am testing on an API 21 emulator if that helps.

**Comment by Dimezis** [comment]
Ok, seems like I found the cause, will release a fix today/tomorrow

**Comment by searchy2** [comment]
You're amazing! I wish all library creators were as responsive as you are. Not even I'm that good :see_no_evil: 

Have a good weekend and happy Easter! :hatching_chick: :hatched_chick: :rabbit:

**Comment by Dimezis** [comment]
Fixed in 1.6.1.
Btw you should now be able to use a regular dialog in your lib and the blur position will be calculated properly

**Comment by searchy2** [comment]
Thanks you so much! You're amazing :)
